Add new `alt` attribute to the `Featured Product` media settings

albarin opened the pull request:
This PR allows defining an `alt` text for the `Featured Product` block image.

Fixes part of https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/6235

### Testing

How to test the changes in this Pull Request:

1. Create a new page and add a `Featured Product`.
2. On the block settings go to `Media Settings` and check the `Alt text` textarea appears empty.
3. Add an `alt` text, save the block and check it renders on the frontend with the specified `alt` text.
4. Edit the block again, remove the `alt` text and save it.
5. Check the `alt` rendered on the frontend corresponds to the product name.

### Changelog

> Add the alt text control to the Featured Product block media settings

albarin committed:
Add new `alt` attribute to Media Settings

Allow to define an alt text for the Feature Product block

sunyatasattva commented on assets/js/blocks/featured-product/block.js:
```diff
@@ -339,6 +341,30 @@ const FeaturedProduct = ( {
+													) }
+												</ExternalLink>
+												{ __(
+													'Leave empty if the image is purely decorative.',
```
I guess this comment here is misleading. Because when the user leaves it empty, the PHP block tries to default with the short description, so that we try to always have an alt text, and we kinda know in this specific use-case that the image is not “purely decorative”.

I think we should prefill the alt textarea with the short description of the product and either:

1. Leave this comment as it is and if the user leaves it empty, we do not use the short description in the PHP block and respect the user's decision.
2. Change the comment to something like: “Leaving it empty will still use the product short description if available”. And basically override the user's decision.

I would go for the second approach honestly. I think there is not much reason to allow the user to delete the alt text from the featured product block.

We could even be a tad more sophisticated, and default alt-text with the short description only if the “Show description” option is turned off. Otherwise, I believe that screen reader users will have redundant information and the image *could* be considered decorative as it is already described by title+description.

Thoughts?

albarin replied:
Yes, agree, it makes more sense to me to go for the second option as well.

> We could even be a tad more sophisticated, and default alt-text with the short description only if the “Show description” option is turned off.

Not sure about that, I feel like if we do that we are adding some not obvious behavior without explaining it anywhere, and could be confusing for users.

EDIT: Just realized that `short_description` can actually contain HTML, even images, so it might not be a good idea to use it as an `alt`. Should we go for the first option but without prefilling (or maybe prefilling with the product name), wdyt?

sunyatasattva replied:
> Just realized that short_description can actually contain HTML, even images, so it might not be a good idea to use it as an alt. Should we go for the first option but without prefilling (or maybe prefilling with the product name), wdyt?

Interesting. Yea, then the first option seems the way to go, considering that it is also similar to how other blocks with ALT do it.

It's a bit sad we can't find a sensible default, because I know that most users won't even bother adding an ALT attribute. Maybe the default should just be the title, then? Or something like “Picture of PRODUCT_TITLE”?

albarin replied:
Yep, to me, using product title as default is what makes more sense.

sunyatasattva commented on assets/js/blocks/featured-product/block.js:
```diff
@@ -339,6 +341,30 @@ const FeaturedProduct = ( {
+										value={ attributes.alt }
+										onChange={ ( newAlt ) => {
+											setAttributes( { alt: newAlt } );
+										} }
```
Super nitpick, but why not:

```suggestion
										onChange={ ( alt ) => {
											setAttributes( { alt } );
										} }
```

albarin committed:
Simplify on change

albarin committed:
Improve alt description and use name instead of short description

Short description can contain html and images, so it's not adequate to use on the alt attribute

sunyatasattva reviewed (approved):
Nice!
